tests/test_check_stac_metadata_utils_logging.py

```diff
@@ -101,9 +103,10 @@ def should_log_non_s3_url_prefix_validation() -> None:
     with patch("geostore.check_stac_metadata.utils.LOGGER.error") as logger_mock, patch(
         "geostore.check_stac_metadata.utils.processing_assets_model_with_meta"
     ):
-        STACDatasetValidator(
-            hash_key, url_reader, MockAssetGarbageCollector(), MockValidationResultFactory()
-        ).run(metadata_url)
+        with raises(InvalidSTACRootTypeError):
```

Comment:
If I merged the `raises` statement up with the update with, it will not reach to the below `assest_any_call  ` condition below.

Reply:
TIL: [`with` does *not* create a scope](https://stackoverflow.com/a/45100308/96588). So you can move `raises(InvalidSTACRootTypeError)` into the top `with` statement if you dedent the `logger_mock.assert_any_call`, like this:

```python
with patch(…) as logger_mock, raises(InvalidSTACRootTypeError):
    STACDatasetValidator(…).run(metadata_url)
logger_mock.assert_any_call(…)
```